### ISSUE TITLE
Loss function for dnorm in predcls

### ISSUE THREAD

**Issue by jkli-aa** [bknyaz/sgg#8]
![image](https://user-images.githubusercontent.com/84705442/165981812-1981daa1-d580-4001-8ddc-8a48b59662b7.png)

**Comment by jkli-aa** [comment]
I got it. The change for $(FG+BG)/FG$ will enlarge the loss of large graph. 
Sorry to bother you.

**Comment by bknyaz** [comment]
Hi, I'm glad you've resoved this issue. You are right in your last comment, but I just wanted to add that we tested the variant when the edge loss weight is fixed for all batches (equivalent to adjusting the learning rate as you mentioned). Such a variant also worked well in some cases (see the paper for details). The weight of our edge loss adapts to each batch of images/scene graphs and does not require tuning for a new dataset (where scene graphs may have different sparsity). So these properties of our loss can be helpful in practice.

**Comment by jkli-aa** [comment]
> 

It really addressed my concerns. Thank you for your kind reply.
